lib/OpenQA/Worker/Common.pm

```diff
@@ -368,8 +368,9 @@ sub call_websocket {
             }
             else {
                 $ws = undef;
-                if ($tx->completed) {
-                    call_websocket($ua_url->parse($tx->res->headers->location));
+                if (my $location_header = ($tx->completed ? $tx->res->headers->location : undef)) {
+                    print "Following ws redirection to: $location_header\n";
+                    call_websocket($ua_url->parse($location_header));
```

Comment:
I liked the old version more. It takes the same amount of lines (assuming you add the print to the old version) and it's easier to read for me.